### PR TITLE
feat: Add and Remove kernels, add with --copy-defaults

### PR DESCRIPTION
**Pull request by spetrosi**
Enhancement:

1. Add an ability to add and remove kernels. To do that, set `state` to either `present` or `absent`.
2. When adding kernels, you can set `copy_defaults: true` to copy arguments from the default kernel.
3. Remove the ability to provide lists with kernel keys, now each kernel key can contain a single dict. This makes it consistent with other roles. You can use `kernel: ALL` to modify all kernels at once.

### CODE REVIEW

**Comment by spetrosi** [comment]
[citest]

**Comment by spetrosi** [comment]
[citest]

**Comment by spetrosi** [comment]
[citest]

**Comment by spetrosi** [comment]
[citest]

**Comment by spetrosi** [comment]
[citest]

**Comment by spetrosi** [comment]
[citest]

**Comment by spetrosi** [comment]
[citest]

**Comment by spetrosi** [comment]
[citest]

**Comment by spetrosi** [comment]
[citest]

**Comment by spetrosi** [comment]
[citest]

**Comment by spetrosi** [comment]
@richm it's now ready for your review, please take a look.
I do not know how to use mock in unit tests to test Ansible's module method. That's why bootloader_settings.py uses a weird format of running commands, it calls a function to return a cmd to run, or an error to return, and then does the action in the main `run_module` func. I will try to refactor this, but the role's functionality will stay the same.

File: tests/tests_add_rm.yml [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+    # Images in CI might have a grub timeout set to a different other than the
+    # default 5 value.
+    # In this case, the above invocation require handlers to be flushed.
+    - name: Flush handlers
+      meta: flush_handlers
+
+    - name: Set fact with the default kernel to use for clones
+      set_fact:
+        test_kernel: "{{ bootloader_facts | rejectattr('initrd', 'undefined')
+          | selectattr('default', 'equalto', True) | first }}"
```
Comment (richm) [inline tests/tests_add_rm.yml:26]:
```suggestion
          | selectattr('default') | first }}"
```
https://jinja.palletsprojects.com/en/2.9.x/templates/#selectattr
"If no test is specified, the attribute’s value will be evaluated as a boolean."

File: tests/tests_add_rm.yml [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+      vars:
+        bootloader_gather_facts: true
+      include_role:
+        name: linux-system-roles.bootloader
+
+    - name: Verify settings
+      vars:
+        __default_args: "{{
+          (bootloader_facts | selectattr('title', 'defined') |
+          selectattr('default', 'equalto', True) |
```
Comment (richm) [inline tests/tests_add_rm.yml:77]:
```suggestion
          selectattr('default') |
```

File: tests/tests_settings.yml [hunk 3]
```diff
@@ -14,28 +14,28 @@
...
         default_bootloader: "{{
-          (bootloader_facts | selectattr('index', 'search', '0') | first).kernel
-          }}"
-      shell: >-
-        set -euo pipefail;
-        grubby --info=DEFAULT;
-        grubby --info=DEFAULT |
-        grep -P
-        'kernel=("|){{ default_bootloader }}("|)'
+          (bootloader_facts | selectattr('default', 'equalto', True) |
```
Comment (richm) [inline tests/tests_settings.yml:26]:
```suggestion
          (bootloader_facts | selectattr('default') |
```

**Comment by richm** [comment]
There are a lot of codeql `Variable defined multiple times` warnings - would be nice to fix or suppress them.

**Comment by spetrosi** [comment]
[citest]

**Review by richm — APPROVED** [review]
ready to merge?

**Comment by spetrosi** [comment]
[citest]

**Comment by spetrosi** [comment]
[citest]

**Pull request merged**